.github/workflows/manual-cherry-pick.yaml

```diff
@@ -0,0 +1,72 @@
+# Copyright 2023 The Cobalt Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Usage: Go to
+# https://https://github.com/youtube/cobalt_sandbox/actions/workflows/manual-cherry-pick.yaml
+# and click "Run Workflow." Leave "Use Workflow From" set to "main", then
+# input the branch name and paste the cherry-pick commit and click Run. A PR
+# will be created.
+
+name: Release Branch Cherrypick
+on:
+  workflow_dispatch:
+    inputs:
+      # We use this instead of the "run on branch" argument because GitHub looks
+      # on that branch for a workflow.yml file, and we'd have to cherry-pick
+      # this file into those branches.
+      release_branch:
+        description: 'Release branch name (e.g. 23.lts.1+)'
+        required: true
+        type: string
+      git_commit:
+        description: 'Git commit to cherry-pick'
+        required: true
+        type: string
+
+jobs:
+  cherrypick:
+    name: Cherrypick to ${{ github.event.inputs.release_branch}} - ${{ github.event.inputs.git_commit }}
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      RELEASE_BRANCH: ${{ github.event.inputs.release_branch }}
+      COMMIT_HASH: ${{ github.event.inputs.git_commit }}
+      REPOSITORY: ${{ github.repository }}
+      GITHUB_REF: ${{ github.ref }}
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+      with:
+        ref: $RELEASE_BRANCH
```

Comment:
I don't think ref is getting set to $RELEASE_BRANCH here, there might be two reason why:

1. Have to evaluate env vars like so: ${{env.RELEASE_BRANCH}} (use GitHub Actions syntax vs bash).
2. It's possible, but I am not entirely sure, that RELEASE_BRANCH env var is not set yet when referenced in this particular step (vs being available in commands executed in step itself).  If that is the case you will have to set ref to ${{github.event.inputs.release_branch}}. I would try #1 and if that doesn't work then #2 should work.